### PR TITLE
chore: use Github App for authentication

### PR DESCRIPTION
**Pull request by ethankhall**
The new script will create an `INSTALLATION_ACCESS_TOKEN` scoped to this repo, and only valid for 1 hour.

This will allow the semantic version step to no longer require a human's identity.

### CODE REVIEW

**Review by grxy — CHANGES_REQUESTED** [review]
Can you add comments to the script explaining what is going on?

**Comment by ethankhall** [comment]
Yes! Sorry for the delay.

File: .circleci/create_release.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+--header 'Accept: application/vnd.github.v3+json' \
+--header "Authorization: Bearer ${signed_content}.${sig}")")
+
+## Get the repo ID
+repo_id=$(jq .id <<< "$(curl -g -s \
+-H "Accept: application/vnd.github.v3+json" \
+--header "Authorization: token $unscoped_token" \
+"https://api.github.com/repos/$repo")")
+
+## Finally geneate the token from the Installation ID that is scopped to juse the repo
```
Comment (grxy) [inline .circleci/create_release.sh:52]: Typo: `juse` => `just`

File: .circleci/create_release.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+token=$(jq .token -r <<< "$(curl --location -g -s \
+--request POST "https://api.github.com/app/installations/${installation_id}/access_tokens" \
+--header 'Accept: application/vnd.github.v3+json' \
+--header "Authorization: Bearer ${signed_content}.${sig}" \
+-d "{\"repository_ids\":[$repo_id]}" )")
+
+## Export the token from semantic-release
+export GH_TOKEN="$token"
+
+## Exec into semmantic release w/ GH_TOKEN set
```
Comment (grxy) [inline .circleci/create_release.sh:62]: Typo: `semmantic` => `semantic`

**Review by grxy — COMMENTED** [review]
Everything else looks good. Will approve after a couple small typos are fixed.

**Review by grxy — APPROVED** [review]

**Pull request merged**